File: src/main/java/jkmau5/alternativeenergy/world/blocks/AltEngBlocks.java
package jkmau5.alternativeenergy.world.blocks;

import cpw.mods.fml.common.registry.GameRegistry;
import jkmau5.alternativeenergy.world.item.ItemBlockPowerStorage;
import net.minecraft.block.Block;
import net.minecraft.block.material.Material;

/**
 * Author: Lordmau5
 * Date: 21.08.13
 * Time: 16:55
 * You are allowed to change this code,
 * however, not to publish it without my permission.
 */
public class AltEngBlocks {

    public static Block blockPowerCable;
    public static Block blockPowerStorage;
    public static Block blockConveyor;

    public static void init() {
        blockPowerCable = new BlockPowerCable(Material.iron);
        blockPowerStorage = new BlockPowerStorage();
        blockConveyor = new BlockConveyor();

        GameRegistry.registerBlock(AltEngBlocks.blockPowerStorage, ItemBlockPowerStorage.class, "powerStorage");
        GameRegistry.registerBlock(AltEngBlocks.blockPowerCable, "powerCable");
        GameRegistry.registerBlock(AltEngBlocks.blockConveyor, "conveyor");
    }
}

File: src/main/scala/alternativemods/alteng/blocks/AltEngBlocks.scala
package alternativemods.alteng.blocks

import cpw.mods.fml.common.registry.GameRegistry
import net.minecraft.block.material.Material
import alternativemods.alteng.blocks.energy.{BlockEnergyPassthrough, BlockEnergyConsumer}
import alternativemods.alteng.items.itemblocks.ItemBlockConveyor
import net.minecraftforge.fluids.{FluidRegistry, Fluid}

object AltEngBlocks {

  var blockConveyor: BlockConveyor = _
  var blockEnergyStorage: BlockEnergyPassthrough = _
  var blockEnergyConsumer: BlockEnergyConsumer = _
  var blockLiquidEnergy: BlockLiquidEnergy = _

  var fluidLiquidEnergy: Fluid = _

  def load(){
    this.fluidLiquidEnergy = new Fluid("altEng.liquidEnergy")

    FluidRegistry.registerFluid(this.fluidLiquidEnergy)

    blockConveyor = new BlockConveyor(Material.iron)
    blockEnergyStorage = new BlockEnergyPassthrough(Material.iron)
    blockEnergyConsumer = new BlockEnergyConsumer
    blockLiquidEnergy = new BlockLiquidEnergy(this.fluidLiquidEnergy)

    fluidLiquidEnergy.setBlock(blockLiquidEnergy)

    GameRegistry.registerBlock(blockConveyor, classOf[ItemBlockConveyor], "conveyor")
    GameRegistry.registerBlock(blockEnergyStorage, "energyPassthrough")
    GameRegistry.registerBlock(blockEnergyConsumer, "energyConsumer")
    GameRegistry.registerBlock(blockLiquidEnergy, "liquidEnergy")
  }
}
